packages/core/core/src/Transformation.js

```diff
@@ -435,6 +436,16 @@ export default class Transformation {
     let resultingAssets = [];
     let finalAssets = [];
     for (let transformer of pipeline.transformers) {
+      let measurement = applicationProfiler.createMeasurement(
```

Comment:
Here also? I am thinking that in the default case (profiling is not enabled), calls to `createMeasurement` should be a no-op, including skipping the allocation of new argument bags. I'm aware that this feels a bit like premature optimization; reasonable folks can disagree.

Reply:
I haven't done any significant perf measurement, but given most "prod" builds would be running with the profiler disabled avoiding unnecessary allocations / GC overhead seems like a reasonable idea.